modules/pytket-qiskit/pytket/extensions/qiskit/backends/ibm.py

```diff
@@ -345,28 +368,45 @@ def process_circuits(
     ) -> List[ResultHandle]:
         """
         See :py:meth:`pytket.backends.Backend.process_circuits`.
-        Supported kwargs: none.
+        Supported kwargs: `postprocess`.
         """
         if n_shots is None or n_shots < 1:
             raise ValueError("Parameter n_shots is required for this backend")
         handle_list = []
         for chunk in itertools.zip_longest(*([iter(circuits)] * self._max_per_job)):
             filtchunk = list(filter(lambda x: x is not None, chunk))
+
             if valid_check:
                 self._check_all_circuits(filtchunk)
-            qcs = [tk_to_qiskit(tkc) for tkc in filtchunk]
+
+            postprocess = kwargs.get("postprocess", False)
+
+            qcs, ppcirc_strs = [], []
+            for tkc in filtchunk:
+                if postprocess:
+                    c0, ppcirc = prepare_circuit(tkc, allow_classical=False)
+                    ppcirc_rep = ppcirc.to_dict()
+                else:
+                    c0, ppcirc_rep = tkc, None
+                qcs.append(tk_to_qiskit(c0))
+                ppcirc_strs.append(str(ppcirc_rep))
             qobj = assemble(qcs, shots=n_shots, memory=self._config.memory)
             if self._MACHINE_DEBUG:
                 handle_list += [
-                    ResultHandle(_DEBUG_HANDLE_PREFIX + str((c.n_qubits, n_shots)), i)
+                    ResultHandle(
+                        _DEBUG_HANDLE_PREFIX + str((c.n_qubits, n_shots)),
+                        i,
+                        ppcirc_strs[i],
```

Comment:
would prefer to use `zip` over `enumerate`

Reply:
Don't we we need enumerate to get the index `i` (second term)?

Reply:
Ah yeah true, you can do `enumerate(zip())` but this is fine